main/config.c

```diff
@@ -2746,12 +2751,47 @@ int ast_config_text_file_save2(const char *configfile, const struct ast_config *
 		/* reset all the output flags in case this isn't our first time saving this data */
 		incl->output = 0;
 
-		if (!incl->exec) {
-			/* now make sure we have write access to the include file or its parent directory */
+		if (incl->include_type <= 0) {
+			/* Not a #exec */
 			make_fn(fn, sizeof(fn), incl->included_file, configfile);
-			/* If the file itself doesn't exist, make sure we have write access to the directory */
-			if (!is_writable(fn)) {
-				return -1;
+			if (incl->include_type == 0) {
+				/* A #include. Make sure we have write access to the include file or
+				   its parent directory. If the file itself doesn't exist, make sure
+				   we have write access to the directory */
+				if (!is_writable(fn)) {
+					return -1;
+				}
+			} else {
+				int ret = 0;
+				int glob_ret;
+				glob_t globbuf;
+
+				/* A #tryinclude. Make sure we have write access to each of the
```

Comment:
I would try moving as much of this globbuf stuff as you can into a separate function as you can (I see there's some more of it further down as well)

Reply:
Done